apps/src/music/player/sequencer/Simple2Sequencer.ts

```diff
@@ -341,8 +347,12 @@ export default class Simple2Sequencer extends Sequencer {
     }
     const currentFunction = this.functionMap[currentFunctionId];
 
-    currentFunction.playbackEvents.push(event);
-    this.updateMeasureForPlayByLength(event.length);
+    const uniqueEventKey = `${event.id}-${event.when}`;
+    if (!this.uniqueEvents.has(uniqueEventKey)) {
```

Comment:
Nice!

Reply:
Neat solution! Should we short circuit this at the beginning of the function so we don't increment the event count for duplicate events? Or do we still want to do that for rendering purposes?

Reply:
Good idea.  Seems alright to short circuit.